doc/dev.rst

```diff
@@ -51,80 +51,100 @@ account the probable desires of the recipient as well.
+  the PR is ready to be merged once tests and code review pass. In-progress
+  PRs headed in that direction should also be in this state (i.e., the trigger
+  for review and possible merge is the review request, not a draft to
+  ready-for-review transition).
 
-**Address a single concern.** When possible, issues and PRs should address
+* *Draft*. This means not ready for merge even if tests and review pass.
+  (GitLab would indicate this with a :code:`WIP:` prefix in the title.)
+
+**Address a single concern.** When practical, issues and PRs should address
 completely one self-contained change. If there are multiple concerns, make
 separate issues and/or PRs. For example, PRs should not tidy unrelated code,
 and non-essential complications should be split into a follow-on issue.
+However, sometimes one PR addresses several related issues, which is fine.
 
 **Documentation and tests first.** The best practice for significant changes
 is to draft documentation and/or tests first, get feedback on that, and then
 implement the code. Reviews of the form "you need a completely different
 approach" are no fun.
 
-**Tests must pass.** PRs will not be merged until they pass the tests. While
-this most saliently includes CI, the tests should also pass on your
-development box as well as all relevant clusters (if appropriate for the
-changes).
+**CI must pass.** PRs will usually not be merged until they pass CI, with
+exceptions if the failures are clearly unconnected and we are confident they
+aren’t masking a real issue. If appropriate, tests should also pass on
+relevant supercomputers.
 
-**No close keywords in PRs.** While GitHub will interpret issue-closing
-keywords (variations on `"closes", "fixes", and "resolves"
+**Use close keywords in PRs.** Use the issue-closing keywords (variations on
+`"closes", "fixes", and "resolves"
 <https://help.github.com/en/articles/closing-issues-using-keywords>`_) in PR
-descriptions, don’t use this feature, because often the specific issues a PR
-closes change over time, and we don’t want to have to edit the description to
-deal with that. We also want this information in only one place (the commit
-log). Instead, use “addresses”, and we’ll edit the keywords into the commit
-message(s) at merge time if needed.
+descriptions to link it to the relevant issue(s). If this changes, edit the
+description to add/remove issues.
 
-**PR review procedure.** When your draft PR is ready for review — which may or
-may not be when you want it considered for merging! — do one or both of:
+**PR review procedure.** When your PR is ready for review — which may or may
+not be when you want it considered for merging! — do this:
 
-* Request review from the person(s) you want to look at it. If you think it
-  may be ready for merge, that should include the project lead. The purpose of
-  requesting review is so the person is notified you need their help.
+#. Request review from the person(s) you want to look at it. The purpose of
```

Comment:
Is it worth briefly mentioning how to choose someone to review your code? I could see the prospect of asking one of us for a code review to be daunting for some new external contributors, being a bit more explicit might help with that.